services/admin/src/admin/routes/num_dataset_infos_by_builder_name.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+from typing import Optional
+
+from libcommon.simple_cache import CachedResponseDocument
+from starlette.requests import Request
+from starlette.responses import Response
+
+from admin.authentication import auth_check
+from admin.utils import (
+    AdminCustomError,
+    Endpoint,
+    UnexpectedError,
+    get_json_admin_error_response,
+    get_json_ok_response,
+)
+
+
+def create_num_dataset_infos_by_builder_name_endpoint(
+    max_age: int,
+    external_auth_url: Optional[str] = None,
+    organization: Optional[str] = None,
+    hf_timeout_seconds: Optional[float] = None,
+) -> Endpoint:
+    async def usage_endpoint(request: Request) -> Response:
+        try:
+            logging.info("/num-dataset-infos-by-builder-name")
+
+            # if auth_check fails, it will raise an exception that will be caught below
+            auth_check(
+                external_auth_url=external_auth_url,
+                request=request,
+                organization=organization,
+                hf_timeout_seconds=hf_timeout_seconds,
+            )
+            num_datasets_infos = CachedResponseDocument.objects(http_status=200, kind="dataset-info").count()
+            num_parquet_datasets_infos = CachedResponseDocument.objects(
+                http_status=200, kind="dataset-info", content__dataset_info__default__builder_name="parquet"
```

Comment:
Maybe all these requests should be inside simple_cache.py and also parameterize the builder_name or maybe use a method that returns summarized information (count of records grouped by builder_name).
Also, If I understand well, are you querying the builder_name for the "default" config? If so, what about the datasets that don't have that config?

Reply:
> count of records grouped by builder_name

I understand that each dataset with its own script will have a different builder name, so it makes sense to only check a list of reserved builder names.

We could optimize the code a bit, but maybe don't spend too much time in this, as it's only for our own admin purpose

Reply:
Ah yes indeed ! It's because I started working on this endpoint to do something else first and reverted it - will remove

Reply:
> Also, If I understand well, are you querying the builder_name for the "default" config? If so, what about the datasets that don't have that config?

Usually datasets with a builder name like parquet/csv etc have the "default" config (<0.1% don't) and it simplifies the mongo request. Alternatively I could query the config-level cache though the query is a bit longer

> We could optimize the code a bit, but maybe don't spend too much time in this, as it's only for our own admin purpose

Yes that's what I had in mind